crates/extensions/c8y_firmware_manager/src/message.rs

```diff
@@ -98,11 +97,17 @@ impl TryFrom<&MqttMessage> for FirmwareOperationResponse {
 
     fn try_from(message: &MqttMessage) -> Result<Self, Self::Error> {
         let topic = &message.topic.name;
-        let child_id = get_child_id_from_child_topic(topic).ok_or(
-            FirmwareManagementError::InvalidTopicFromChildOperation {
-                topic: topic.into(),
-            },
-        )?;
+        let child_id = match topic.split('/').collect::<Vec<&str>>()[..] {
+            // This plugin is still listening to the deprecated `tedge` command topic
+            ["tedge", child_id, "commands", "res", "firmware_update"] if !child_id.is_empty() => {
+                child_id.to_string()
+            }
+            _ => {
+                return Err(FirmwareManagementError::InvalidTopicFromChildOperation {
+                    topic: topic.into(),
+                })
+            }
+        };
```

Comment:
note: I think this not only removes the deprecated function, but also fixes the wrong behaviour

Previously `get_child_id_from_child_topic` just extracted child id, but the topic itself wasn't checked, so it could be anything.

Reply:
Indeed, this also fixes the behavior and I failed to missed that trying at first to use one of the `tedge_api::MqttSchema` method!